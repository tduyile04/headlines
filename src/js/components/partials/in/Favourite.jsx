import React from 'react';
import { withRouter } from 'react-router-dom';
import HeadlineAction from '../../../actions/HeadlineAction';

class Favourite extends React.Component {
  constructor() {
    super();
    this.showFullArticle = this.showFullArticle.bind(this);
  }

  /**
   * Creates an action to display the full article
   * @param {any} url address of the article to be scraped
   * @memberof Favourite
   */
  showFullArticle(url) {
    HeadlineAction.showFullArticle(url);
    this.props.history.push('/article/detail');
  }

  render() {
    const { favourite } = this.props;
    return (
      <div>
        <section>
          <div className="col s12 m6">
            <div className="card grey darken-3">
              <div className="card-content white-text">
                <span className="card-title">{favourite.title}</span>
                <p>{favourite.description}</p>
              </div>
              <div className="card-action" style={{ height: '70px' }}>
                <button
                  className="btn right blue lighten-1"
                  onClick={() => this.showFullArticle(favourite.url)}
                >
                  View Articles
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

export default withRouter(Favourite);
