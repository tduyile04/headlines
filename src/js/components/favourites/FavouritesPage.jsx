import React from 'react';
import Favourite from '../partials/in/Favourite.jsx';
import NavBar from '../partials/in/NavBar.jsx';
import Footer from '../partials/in/Footer.jsx';


/**
 * Page displaying all the articles added to the collection
 * @returns {component}
 */
function FavouritesPage() {
  const user = localStorage.getItem('userProfile');
  const stringTypeFavourites = localStorage.getItem('saved-articles');
  let Favourites;
  try {
    Favourites = JSON.parse(stringTypeFavourites);
  } catch (e) {
    Favourites = '';
  }
  return (
    <div>
      <NavBar user={user} />
      <h4
        className="orange-text text-accent-1"
        style={{ paddingLeft: '10px' }}
      >
        Archives
      </h4>
      <div className="row">
      {
        !Favourites && (
          <div>
            <h4 className="white-text">No Favourites Added yet</h4>
          </div>
        )
      }
      {
        Favourites.map((favourite, index) => {
          return <Favourite key={index} favourite={favourite} />;
        }, this)
      }
      </div>
      <Footer />
    </div>
  );
}

export default FavouritesPage;
