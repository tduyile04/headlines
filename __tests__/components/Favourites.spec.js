import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Favourite from '../../src/js/components/partials/in/Favourite.jsx';
import FavouritesPage from
'../../src/js/components/favourites/FavouritesPage.jsx';

describe('The Favourite component', () => {
  beforeEach(() => {
    const data = [{
      title: 'harry porter 10',
      url: 'http://localhost:8080/favourites',
      description: 'the story of harry porter'
    }];
    localStorage.setItem('saved-articles', JSON.stringify(data));
  });
  it('should display the same view for each render', () => {
    const wrapper = shallow(<Favourite />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render the same FavouritesPage component every time', () => {
    const wrapper = shallow(<FavouritesPage />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render the same Favourite component every time', () => {
    const favourite = {
      title: 'random title',
      description: 'a random description',
      url: 'http://localhost:8080/favourites'
    };
    const tree = renderer.create(
      <MemoryRouter><Favourite favourite={favourite} /></MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
