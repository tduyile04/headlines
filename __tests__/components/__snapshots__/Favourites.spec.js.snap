// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favourite /> should display the same view for each render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route
    render={[Function]}
/>,
  "nodes": Array [
    <Route
      render={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <withRouter(Favourite) />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Route
          render={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <Route
          render={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(Favourite) />,
}
`;

exports[`<Favourite /> should render the same <Favourite /> 1`] = `
<div>
  <section>
    <div
      className="col s12 m6"
    >
      <div
        className="card grey darken-3"
      >
        <div
          className="card-content white-text"
        >
          <span
            className="card-title"
          >
            random title
          </span>
          <p>
            a random description
          </p>
        </div>
        <div
          className="card-action"
          style={
            Object {
              "height": "70px",
            }
          }
        >
          <button
            className="btn right blue lighten-1"
            onClick={[Function]}
          >
            View Article
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Favourite /> should render the same <FavouritesPage /> each time 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <withRouter(NavBar)
        user={null}
    />
    <h4
        className="orange-text text-accent-1"
        style={
            Object {
                "paddingLeft": "10px",
              }
        }
    >
        Favourites
    </h4>
    <div
        className="row"
    >
        <withRouter(Favourite)
            favourite={
                Object {
                    "description": "the story of harry porter",
                    "title": "harry porter 10",
                    "url": "http://localhost:8080/favourites",
                  }
            }
        />
    </div>
    <Footer />
</div>,
  "nodes": Array [
    <div>
      <withRouter(NavBar)
            user={null}
      />
      <h4
            className="orange-text text-accent-1"
            style={
                  Object {
                        "paddingLeft": "10px",
                      }
            }
      >
            Favourites
      </h4>
      <div
            className="row"
      >
            <withRouter(Favourite)
                  favourite={
                        Object {
                              "description": "the story of harry porter",
                              "title": "harry porter 10",
                              "url": "http://localhost:8080/favourites",
                            }
                  }
            />
      </div>
      <Footer />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FavouritesPage />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <withRouter(NavBar)
                    user={null}
          />
          <h4
                    className="orange-text text-accent-1"
                    style={
                              Object {
                                        "paddingLeft": "10px",
                                      }
                    }
          >
                    Favourites
          </h4>
          <div
                    className="row"
          >
                    <withRouter(Favourite)
                              favourite={
                                        Object {
                                                  "description": "the story of harry porter",
                                                  "title": "harry porter 10",
                                                  "url": "http://localhost:8080/favourites",
                                                }
                              }
                    />
          </div>
          <Footer />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <withRouter(NavBar)
                    user={null}
          />
          <h4
                    className="orange-text text-accent-1"
                    style={
                              Object {
                                        "paddingLeft": "10px",
                                      }
                    }
          >
                    Favourites
          </h4>
          <div
                    className="row"
          >
                    <withRouter(Favourite)
                              favourite={
                                        Object {
                                                  "description": "the story of harry porter",
                                                  "title": "harry porter 10",
                                                  "url": "http://localhost:8080/favourites",
                                                }
                              }
                    />
          </div>
          <Footer />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FavouritesPage />,
}
`;
